pkg/aws/actuator/actuator.go

```diff
@@ -840,7 +840,30 @@ func (a *AWSActuator) tagUser(logger log.FieldLogger, awsClient minteraws.Client
 		})
 	}
 
-	_, err := awsClient.TagUser(&iam.TagUserInput{
+	// appending the tags present in the Infrastructure CR
+	infra, err := utils.GetInfrastructure(a.Client)
+	if err != nil {
+		logger.WithError(err).Error("unable to get the infrastructure object")
+	} else {
+		// The ResourceTags field would be migrated from the Status field to the Spec field.
+		// Till the field is deprecated, the ResourceTags present in the Spec field takes the priority
+		var userTags []configv1.AWSResourceTag
+		if len(infra.Spec.PlatformSpec.AWS.ResourceTags) == 0 {
+			userTags = infra.Status.PlatformStatus.AWS.ResourceTags
+		} else {
+			userTags = infra.Spec.PlatformSpec.AWS.ResourceTags
+		}
```

Comment:
Why are we referencing both status and spec? What is the relationship between status and spec. The comment indicates that information would migrate "from the Status field to the Spec field" which sounds like the reverse flow for how information is typically updated in kubernetes objects.

Reply:
I actually meant to convey the following 

> Saving in the status field means it's not included in Velero backups, which is a crucial feature for customers and Day2.
> Thus the status.resourceTags field should be deprecated in favour of a newly created spec.resourceTags with the same content. The installer should only populate the spec, consumers of the infrastructure CRD must favour the spec over the status definition if both are supplied, otherwise the status should be honored and a warning shall be issued.

which is part of this [RFE](https://issues.redhat.com/browse/RFE-2012).

Modified the code to fetch the info from the Spec.

Reply:
In what situation would status be populated, but not spec? I guess what I might really be asking is: If this repo today doesn't support reading from status, and spec is the "one true location for tag info", why bother reading status at all?

Reply:
Nevermind, it looks like the latest version of the PR only reads from spec.

Reply:
> In what situation would status be populated, but not spec? I guess what I might really be asking is: If this repo today doesn't support reading from status, and spec is the "one true location for tag info", why bother reading status at all?

As of today, all the other operators supporting this feature of user tags are reading from the Status field of the Infrastructure resource. Initially, I added code to read from any of the field and if both are present, the Spec takes the priority according to the RFE. Anyway, if the dependent PR mentioned above is merged, the Infrastructure resource would have the resourceTags in the spec field.